src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -509,6 +513,107 @@ private OidcConfiguration buildOidcConfiguration() {
         return conf;
     }
 
+    private static void filterJwsAlgorithms(OIDCProviderMetadata oidcProviderMetadata) {
```

Comment:
Looks dangerous, what if we're using an immutableList? All the `set*JW*Algs` methods are public